agent-process-launcher/src/com/thoughtworks/go/agent/AgentOutputAppender.java

```diff
@@ -75,7 +75,7 @@ private RollingFileAppender rollingAppender(String file) throws IOException {
     }
 
     private static String getEffectiveLogDirectory(String file) {
-        String logDir = System.getenv(LOG_DIR);
+        String logDir = new SystemEnvironment().getPropertyImpl("go.agent.log.dir");
```

Comment:
The property is set at the time of initialization in AgentLauncherImpl. Just verified that it works. Will check for linux